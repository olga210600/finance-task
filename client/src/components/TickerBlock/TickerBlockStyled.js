import styled from "styled-components";

export const BlockWrapper = styled.div`
  width: 30%;
  min-width: 300px;
  height: 200px;
  border-radius: 10px;
  margin: 20px;
  box-sizing: border-box;
  padding: 17px 13px ;
  box-shadow:rgb(0 0 0 / 20%) 0px 2px 1px -1px, rgb(0 0 0 / 14%) 0px 1px 1px 0px, rgb(0 0 0 / 12%) 0px 1px 3px 0px;
  //box-shadow: 0 0 5px 0 rgba(0, 0, 0, 0.2);
  position: relative;
`
export const Ticker = styled.div`
  font-size: 27px;
  font-family: sans-serif;
  box-sizing: border-box;
  margin:  10px;
  color: #593c4a;
 
  //margin-top: 20px;
  font-weight: bold;
`
export const Wrap = styled.div`
  display: flex;
`

export const Abbreviation = styled.div`
  font-size: 15px;
  font-family: sans-serif;
  margin: 10px;
`

export const Percent = styled(Abbreviation)`&& {
  font-size: 15px;
}`

export const Price = styled.div`
  font-size: 23px;
  font-family: sans-serif;
  margin: 10px;
`

export const PriceDifference = styled(Abbreviation)`&& {
  color: ${({ isNegative }) => isNegative ? '#f01f1f' : '#12b021'};
  font-size: 15px;

}`
export const Logo = styled.div`
 position: absolute;
  left: 200px;
  top: 60px
`;