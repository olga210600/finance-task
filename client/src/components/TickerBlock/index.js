import React from 'react';
import {BlockWrapper, Ticker, Price, Abbreviation, PriceDifference, Percent, Wrap, Logo} from './TickerBlockStyled'
import {getFullNameTicker, checkOfNegativeNumber} from '../../utils/commonFunctions';

const TickerBlock = ({ticker, price, change_percent, prevPrice, logo}) => {
    const priceData = checkOfNegativeNumber(prevPrice, price)

    return (
        <BlockWrapper>
            <Logo>{logo}</Logo>
            <Ticker> {getFullNameTicker(ticker)} </Ticker>
            <Abbreviation>{ticker} </Abbreviation>
            <Wrap>
                <Percent>{change_percent} % </Percent>
                <PriceDifference isNegative={priceData.isNegative}>{priceData.result} </PriceDifference>
            </Wrap>
            <Price>{price} $</Price>
        </BlockWrapper>
    )
}

export default TickerBlock;