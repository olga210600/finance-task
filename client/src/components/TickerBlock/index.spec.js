import TickerBlock from "./index";
import renderer from 'react-test-renderer'

it('renders', () => {
    const tree = renderer.create(<TickerBlock/>).toJSON();
    expect(tree).toMatchSnapshot()
})

