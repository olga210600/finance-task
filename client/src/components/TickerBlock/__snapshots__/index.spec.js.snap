// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="sc-bdvvtL ieGUHk"
>
  <div
    className="sc-iCfMLu gYJOav"
  />
  <div
    className="sc-gsDKAQ gGoBBq"
  >
     
    
     
  </div>
  <div
    className="sc-hKwDye jAYmyr"
  >
     
  </div>
  <div
    className="sc-dkPtRN cgynxk"
  >
    <div
      className="sc-hKwDye sc-eCImPb jAYmyr guOTQf"
    >
       % 
    </div>
    <div
      className="sc-hKwDye sc-gKclnd jAYmyr jPnTxo"
    >
      - No data - 
       
    </div>
  </div>
  <div
    className="sc-jRQBWg fYXUcB"
  >
     $
  </div>
</div>
`;
